src/nat/data_models/span.py

```
@@ -128,10 +128,48 @@ class SpanStatus(BaseModel):
+                         description="The OTel-syle 64-bit span ID of the span.")
+
+    @field_validator("trace_id", mode="before")
+    @classmethod
+    def _validate_trace_id(cls, v: int | str | None) -> int:
+        """Regenerate if trace_id is None; raise an exception if trace_id is invalid;"""
+        if isinstance(v, str):
+            v = uuid.UUID(v).int
+        if isinstance(v, type(None)):
+            v = _generate_nonzero_trace_id()
+        if v == 0:
+            raise Exception("A null trace_id is invalid")
```

Ruff (0.13.2)

Create your own exception

(TRY002)